atlp-rwanda/champs-ec-be
# #187300226  Added HoundCI badges and fix ESLint errors

### **What does this PR do?**

This PR implements the tasks related to configuring ESLint, adding HoundCI badge, and fixing configuration errors. It addresses the requirements outlined in Pivotal ID #187300226.

**Description of Task to be completed:**


1. Added HoundCI badge to enhance code quality monitoring.
2. Fixed some configuration errors.
3.  Set up ESLint for code linting and formatting.
4.  Added lint-staged to run ESLint on staged files before committing.

**How should this be manually tested?**

1.Ensure ESLint is correctly configured by running the lint script (npm run lint) and verifying that code style rules are enforced.
2.Test the Git pre-commit hook by attempting to commit changes. Ensure that ESLint and tests are executed automatically before committing.
3.Check if the HoundCI badge is displayed correctly in the project README.md.

**Screenshots**
![Screenshot 2024-03-30 155022](https://github.com/atlp-rwanda/champs-ec-be/assets/141851169/48af0682-27bd-4708-8f33-0a138b5bc9a6)


**What are the relevant pivotal tracker/Trello stories?**

Pivotal ID #187300226.
